services/QuillLMS/spec/support/shared/unit_assignments_variables.rb

```diff
@@ -37,12 +37,18 @@
   end
 
   def unit_templates_have_a_corresponding_unit?(unit_template_ids)
-    names_from_templates = UnitTemplate.where(id: unit_template_ids).pluck(:name)
-    (Unit.all.map(&:name).flatten & names_from_templates).length == names_from_templates.length
+    UnitTemplate
+      .where(id: unit_template_ids)
+      .pluck(:name)
+      .difference(Unit.pluck(:name))
+      .none?
```

Comment:
This is so much more readable than the old forms of these functions.  Nicely done.